common-test/src/main/scala/org/genivi/sota/data/DeviceGenerators.scala

```diff
@@ -52,20 +54,13 @@ trait DeviceGenerators {
     genConflictFreeDeviceTs(arbitrary[Int].sample.get)
 
   def genConflictFreeDeviceTs(n: Int): Gen[Seq[DeviceT]] = {
-    val names: Seq[DeviceName] =
-      Gen.containerOfN[Seq, DeviceName](n, genDeviceName)
-      .suchThat { c => c.distinct.length == c.length }
-      .sample.get
-    val ids: Seq[DeviceId] =
-      Gen.containerOfN[Seq, DeviceId](n, genDeviceId)
-      .suchThat { c => c.distinct.length == c.length }
-      .sample.get
-
-    val namesG: Seq[Gen[DeviceName]] = names.map(Gen.const(_))
-    val idsG: Seq[Gen[DeviceId]] = ids.map(Gen.const(_))
-
-    namesG.zip(idsG).map { case (nameG, idG) =>
-      genDeviceTWith(nameG, idG).sample.get
+    for {
```

Comment:
Yes, but the version before also produced conflicts. I didn't have any luck trying to eliminate them completely. This version seems clearer however.